### ISSUE TITLE
@Beans in MeterBindersConfiguration should be optional

### ISSUE THREAD

**Issue by michaelom** [micrometer-metrics/micrometer#200]
version: 1.0.0-rc.3

I would like to disable all `MeterBinder`s defined in `MeterBindersConfiguration` (JVM, Uptime, Processor, Logback). Here is an example:
```
    @Bean
    @ConditionalOnMissingBean(JvmMemoryMetrics.class) //   <=== bad
    public JvmMemoryMetrics jvmMemoryMetrics() {
        return new JvmMemoryMetrics();
    }
```
These MeterBinders are conditional beans, yet its very hard to disable them. The only way I could come up with was, to declare a custom NoOp-Implementation:

```
    @Bean
    public UptimeMetrics disabledUptimeMetrics() {
        return new UptimeMetrics() {
            @Override
            public void bindTo(MeterRegistry registry) {
            }
        };
    }
```

~This approach is bad and what's more, it is not working reliably: `MeterBindersConfiguration` is not hooked into Springs AutoConfiguration mechanism (via `spring.factories`).~ edit: This is nonsense. ;-)

I would prefer to disable `MeterBinder`s defined in `MeterBindersConfiguration` using a environment property. 

```
    @Bean
    @ConditionalOnProperty(some.property)
    public JvmMemoryMetrics jvmMemoryMetrics() {
        return new JvmMemoryMetrics();
    }
```

**Comment by checketts** [comment]
Great feedback!

**Comment by spencergibb** [comment]
I think the conditional on missing bean should stay.

**Comment by michaelom** [comment]
That was quick.

> I think the conditional on missing bean should stay.

Works for me, I don't mind.
But I think it's very unlikely someone would subclass `JvmMemoryMetrics` or `ProcessorMetrics`. I would rather disable the beans and implement a custom `MetricsBinder` to measure memory or processor stats.

**Comment by spencergibb** [comment]
:-) I didn't say, the conditional on property shouldn't be added. I think there should be both.

**Comment by checketts** [comment]
Oh, right in case the user has defined their own.

**Comment by jkschneider** [comment]
Fixed with #201. Thanks for the detailed report @michaelom!
